File: testprojects/java/accuracy/domain/indirect/allowedfrom/AccessInstanceVariableIndirectIndirect_VarVarVar.java
package domain.indirect.allowedfrom;

import domain.indirect.intermediate.BackgroundService;

public class AccessInstanceVariableIndirectIndirect_VarVarVar {
	private BackgroundService bckg;
	private String test;

	public AccessInstanceVariableIndirectIndirect_VarVarVar(){
		bckg = new BackgroundService();
	}
		// Indirect-indirect access of a normal attribute 
		public String case4() {
			test = bckg.serviceTwo.serviceOne.name;
			return test;
		}
}		

File: testresources/csharp/benchmark/Domain/Indirect/ViolatingFrom/AccessInstanceVariableIndirectIndirect_VarVarVar.cs
namespace Domain.Indirect.ViolatingFrom
{
	public class AccessInstanceVariableIndirectIndirect_VarVarVar
	{
        private Domain.Indirect.Intermediate.BackgroundService bckg;

		private string test;

		public void Case4()
        {
            bckg = new Domain.Indirect.Intermediate.BackgroundService();
			test = bckg.serviceTwo.serviceOne.name;
		}
	}
}
